astropy/units/format/vounit.py

```diff
@@ -91,7 +91,10 @@ def parse(cls, s, debug=False):
             raise core.UnitsError(
                 "'{}' contains multiple slashes, which is "
                 "disallowed by the VOUnit standard".format(s))
-        result = cls._do_parse(s, debug=debug)
+        with warnings.catch_warnings():
+            # Unsupported warning should be handled by parse_strict in core.py
+            warnings.filterwarnings('error', message=r'.*not supported by the VOUnit standard.*')
+            result = cls._do_parse(s, debug=debug)
```

Comment:
The code here catches a warning and converts it to an exception, but wouldn't it be better to raise an exception to begin with?